lua/nvim-tree/view.lua

```diff
@@ -321,8 +321,19 @@ local function grow()
     max_width = get_width(M.View.max_width) - padding
   end
 
-  for _, l in pairs(lines) do
+  local ns_id = vim.api.nvim_get_namespaces()["NvimTreeExtmarks"]
+  for line_nr, l in pairs(lines) do
     local count = vim.fn.strchars(l)
+    -- also add space for right-aligned icons
+    local extmarks = vim.api.nvim_buf_get_extmarks(0, ns_id, { line_nr, 0 }, { line_nr, -1 }, { details = true })
```

Comment:
This looks like it might be the issue: we're getting extmarks for the current buffer rather than the tree buffer.

Reply:
Thank you, fixed!